recipes/arm-gnu-toolchain/all/conanfile.py

```diff
@@ -0,0 +1,97 @@
+from conan import ConanFile
+from conan.tools.files import get, copy, check_sha256, unzip
+from conan.errors import ConanException
+from urllib.parse import urlparse
+import urllib.request
+import os
+
+
+required_conan_version = ">=1.50.0"
+
+
+class ArmGnuToolchain(ConanFile):
+    name = "arm-gnu-toolchain"
+    license = "GPL-3.0-only"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads"
+    description = ("Conan installer for the GNU Arm Embedded Toolchain")
+    topics = ("gcc", "compiler", "embedded", "ARM", "cortex", "cortex-m",
+              "cortex-m0", "cortex-m0+", "cortex-m1", "cortex-m3", "cortex-m4",
+              "cortex-m4f", "cortex-m7", "cortex-m23", "cortex-m55",
+              "cortex-m35p", "cortex-m33")
+    settings = "os", "arch"
+    short_paths = True
+
+    @property
+    def download_info(self):
+        version = self.version
+        os = str(self.settings_build.os)
+        arch = str(self.settings_build.arch)
+        return self.conan_data.get("sources", {}).get(version, {}).get(os, {}).get(arch)
+
+    @property
+    def license_info(self):
+        version = self.version
+        return self.conan_data.get("sources", {}).get(version, {}).get("license")
+
+    def validate(self):
+        if str(self.settings.os) == "baremetal":
+            return
+        elif not self.download_info:
+            raise ConanException(
+                "This package is not available for this operating system and architecture.")
+
+    def source(self):
+        pass
+
+    def build(self):
+        url = self.download_info["url"]
+        url = url + "?rev=" + self.download_info["rev"]
+        url = url + "&hash=" + self.download_info["hash"]
+        filename = os.path.basename(urlparse(url).path)
+
+        # using urlretrieve over conan.tools.files.get because get results in
+        # a 405 forbidden error whereas urlretrieve does not
+        urllib.request.urlretrieve(url, filename)
```

Comment:
This is problematic, because it will not be cached by the (work in progress) new cache for sources. It will work only for ``conan.tools.files import download/get``. So it is important to check why, and try to solve it, instead of workarounding that in the recipe.